### DISCUSSION TITLE
Usage with custom component

### DISCUSSION THREAD

**Discussion by Me-Phew** [victorgarciaesgi/nuxt-typed-router#116] [Q&A] [answered]
Hi, I need to add a lot of links on my website that look like buttons with icons in them so I have to create a wrapper component over NuxtLink with custom styling. Is this possible without losing the functionality of this module?

**Comment by victorgarciaesgi** [comment, marked as answer]
The `@typed-router` alias export nearly all generated types, so you can play with your component generics. You can look at the bottom of your `.nuxt/typed-router/typed-router.d.ts` for the `NuxtLink` types definitions.
Not supported at the moment to support this natively as the types are pretty complex.

**Comment by KnifeFed** [comment]
I second this. I would love to be able to use it with e.g. `NuxtLinkLocale` from Nuxt i18n etc.
